### PR TITLE
Updating Button component

### PR DESCRIPTION
**Pull request by BlueFrog130**
#35 Updating button component to handle icons.
- updated to allow for use as either an `a` tag or a `button` tag.
- Added `text` prop for alternate styling
- Made all props optional
- added `icon` and `icon-right` prop which accepts an Icon Vue component
- added `icon-color` prop to change icon color

Demo:

vue icon demo

Markdown Usage:
```vue
<script setup>
import CopyIcon from ".vitepress/theme/Components/Icons/CopyIcon.vue";
import ExternalIcon from ".vitepress/theme/Components/Icons/ExternalIcon.vue";
</script>

<BButton color="blue" :icon="CopyIcon" icon-color="white">Normal</BButton>

<BButton color="green" :icon-right="ExternalIcon" link="/">Link</BButton>

<BButton text>Text</BButton>
```

### CODE REVIEW

**Comment by BlueFrog130** [comment]
Required to rename global component to `BButton` (for Bedrock Button) due to naming conflicts for dynamic component. If the button name remains as `Button`, there would be an [infinite reference loop](https://github.com/Bedrock-OSS/bedrock-wiki-vite/blob/71a98d3ac776b535bef406aa0407e362cef4439a/docs/.vitepress/theme/Components/Content/Button.vue#L2).

File: docs/.vitepress/theme/Components/Content/Button.vue [hunk 1]
```diff
@@ -1,14 +1,34 @@
...
+		<div>
+			<slot />
+		</div>
+		<component v-if="iconRight" class="btn-icon" :class="[iconColor]" :is="iconRight" />
+	</component>
 </template>
 
-<script setup lang="ts">
-import { defineProps } from 'vue'
+<script setup="props" lang="ts">
```
Comment (solvedDev) [inline docs/.vitepress/theme/Components/Content/Button.vue:11]: You can revert the change here. `<script setup lang="ts">` is enough. Importing props this way is no longer needed

**Review by solvedDev — CHANGES_REQUESTED** [review]

File: docs/.vitepress/theme/Components/Content/Button.vue (hunk 1 above)
Comment (BlueFrog130) [reply docs/.vitepress/theme/Components/Content/Button.vue]: Done.

Really like the `setup` syntax. Never knew it was a thing until going through code.

**Review by solvedDev — APPROVED** [review]
LGTM
